File: frontend/lib/models/party_stats.dart
import 'package:frontend/models/political_party.dart';

class PartyStats {
  PoliticalParty politicalParty;
  double partyAffectionScore;
  int totalAmountOfProposalsVotedOn;
  double totalAffectionPoints;

  PartyStats({
    required this.politicalParty,
    required this.partyAffectionScore,
    required this.totalAmountOfProposalsVotedOn,
    required this.totalAffectionPoints,
  });

  factory PartyStats.fromJson(Map<String, dynamic> json) {
    return PartyStats(
      politicalParty: PoliticalParty.fromJson(json['politicalParty']),
      partyAffectionScore: json['partyAffectionScore'],
      totalAmountOfProposalsVotedOn: json['totalAmountOfProposalsVotedOn'],
      totalAffectionPoints: json['totalAffectionPoints'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'political_party': politicalParty.toJson(),
      'party_affection_score': partyAffectionScore,
      'total_amount_of_proposals_voted_on': totalAmountOfProposalsVotedOn,
      'total_affection_points': totalAffectionPoints,
    };
  }
}

File: backend/Domain/Entities/PartyStats.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Parlamento.Domain.Entities;

public class PartyStats
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required]
    public PoliticalParty? PoliticalParty { get; set; }

    [Required]
    public double PartyAffectionScore { get; set; }

    [Required]
    public int totalAmountOfProposalsVotedOn { get; set; }

    [Required]
    public double totalAffectionPoints { get; set; }
}
